quickwit/quickwit-indexing/src/actors/merge_planner.rs

```diff
@@ -42,15 +42,37 @@ use crate::MergePolicy;
 /// The merge planner decides when to start a merge task.
 pub struct MergePlanner {
     pipeline_id: IndexingPipelineId,
+
     /// A young split is a split that has not reached maturity
     /// yet and can be candidate to merge operations.
     partitioned_young_splits: HashMap<u64, Vec<SplitMetadata>>,
+    num_young_splits: usize,
+
+    /// This set contains all of the split ids that we "acknowledged".
+    /// The point of this set is to rapidly dismiss redundant `NewSplit` message.
+    ///
+    /// Complex scenarii that can result in the reemission of
+    /// such messages are described in #3627.
+    ///
+    /// At any given point in time, the set must contains at least:
+    /// - young splits (non-mature)
+    /// - splits that are currently in merge.
+    ///
+    /// It also contain other splits, that have gone through a successful
+    /// merge and have been deleted for instance.
+    ///
+    /// We incrementally build this set, by adding new splits to it.
+    /// When it becomes too large, we entirely rebuild it.
+    known_split_ids: HashSet<String>,
```

Comment:
We could get rid of `incarnation_started_at` now that we have this hash set, right?

Reply:
Not really. Without it, we  keep increasing the number of loops that updating the metrics of the merge planner.

Reply:
Overall I feel like it is safer to keep it on both messages.

Reply:
Oh I totally missed the `RefreshMetric` issue... so yes let's keep it for PlanMerge too.

Reply:
It's worth completing the docs of `incarnation_started_at`. Adding something like "Since we recycle mailbox between... [...] taht contains `PlanMerge` or `RefreshMetric` messages" + "Recycling `RefreshMetric` will lead to increasing the number of loops that update the metrics of the merge planner."

Reply:
I need to edit the documentation yes

Reply:
Edited